core/src/main/java/hudson/util/RemotingDiagnostics.java

```diff
@@ -215,6 +215,8 @@ public void doIndex(StaplerResponse rsp) throws IOException {
         @WebMethod(name = "heapdump.hprof")
         public void doHeapDump(StaplerRequest req, StaplerResponse rsp) throws IOException, InterruptedException {
             owner.checkPermission(Jenkins.ADMINISTER);
+            String propertyName = "jdk.management.heapdump.allowAnyFileSuffix";
```

Comment:
wouldn't it be easier to just change the suffix to `.hprof` in line 179?

Reply:
ok, I changed  it.